### ISSUE TITLE
ACRA 4.8.5 Annotation is ignored for ReportingInteractionMode.Dialog

### ISSUE THREAD

**Issue by ChuongPham** [ACRA/acra#405]
I have created the following classes for ACRA:

`public class LocalReportSender implements ReportSender {
        @Override
	public void send(Context context, CrashReportData report) throws ReportSenderException {
        }
}`

`public class LocalReportSenderFactory implements ReportSenderFactory {
	@Override
	public ReportSender create(Context context, ACRAConfiguration config) {
		return new LocalReportSender();
	}
}`

```
@ReportsCrashes(
	buildConfigClass = ApplicationUtil.class,
	mode = ReportingInteractionMode.DIALOG,
	resDialogTitle = R.string.log_heading,
	resDialogText = R.string.log_msg,
	resDialogCommentPrompt = R.string.log_user_comment,
	reportSenderFactoryClasses = { LocalReportSenderFactory.class }
)
public class ApplicationUtil extends Application {
	private static Context mContext;
	private void initialiseAcra() {
		// Initiate ACRA reporting engine.
		ACRA.init(this);
	}
        @Override
       protected void attachBaseContext(Context base) {
           super.attachBaseContext(base);
           initialiseAcra();
	}
}
```

However, when I ran my Android app, instead of getting the ACRA error dialog window, I got the default Android error window instead.

Is this a bug with ACRA 4.8.5 with regard to using annotation?

I have also declared these directives in AndroidManifest.xml file:

```
		<activity
			android:name="org.acra.CrashReportDialog"
	        android:theme="@style/AppCompat.Dialog.Dark.Theme"
        	android:launchMode="singleInstance"
        	android:excludeFromRecents="true"
        	android:finishOnTaskLaunch="true">
		</activity>
		<service
            android:name="org.acra.sender.SenderService"
            android:exported="false"
            android:process=":acra"/>
```

**Comment by william-ferguson-au** [comment]
Yeah, there seems to be something wrong there.
I get the same behaviour.

Want to dig in? I'm not going to get to this for weeks.

**Comment by F43nd1r** [comment]
I could not reproduce this. I can see the correct dialog with your configuration.

By the way: you don't need to declare either of these in your manifest:
`org.acra.CrashReportDialog` is only there for backwards compatibility and not used by default.
`org.acra.sender.SenderService` is already declared in acra's manifest, no need to double it up.

**Comment by william-ferguson-au** [comment]
OK, I could reproduce on an actual device running Android 6.0.1.
@F43nd1r , @ChuongPham want environment were you testing in?

**Comment by F43nd1r** [comment]
Gradle build targeting API 23, run on an API 23 Emulator. The test app is otherwise empty.

**Comment by ChuongPham** [comment]
Mine was Android 5.0 Lollipop Samsung Galaxy S5. I don't get the dialog
with ACRA 4.8.5, just the default crash dialog from the Android platform.
Rolling back to ACRA 4.6.1 with similar config work, however.

**Comment by ChuongPham** [comment]
I spoke too soon!

Rolling back to ACRA 4.61, or even 4.7 will show the error dialog, unlike ACRA 4.8.5 using annotation for same method.

The down side to using pre 4.8 versions is that they aren't compatible with Android API 23 support and build using Android Platform Tools v23.1 and Proguard, since that latter complains:

1) With ACRA 4.6.1, I got an error with Apache's `HttpClient` when building with Proguard. Of course, Android API 23 will only support the use of `HttpURLConnection`.

2) With ACRA 4.7, I got this error regarding the use of Android's notification method when building with Proguard

`[2016-03-27 01:31:23 - MyApp] Proguard returned with error code 1. See console
[2016-03-27 01:31:23 - MyApp] Warning: android.support.v4.app.NotificationCompatBase: can't find referenced method 'void setLatestEventInfo(android.content.Context,java.lang.CharSequence,java.lang.CharSequence,android.app.PendingIntent)' in library class android.app.Notification
[2016-03-27 01:31:23 - MyApp] Warning: android.support.v4.app.NotificationCompatGingerbread: can't find referenced method 'void setLatestEventInfo(android.content.Context,java.lang.CharSequence,java.lang.CharSequence,android.app.PendingIntent)' in library class android.app.Notification
[2016-03-27 01:31:23 - MyApp] Warning: there were 2 unresolved references to library class members.
[2016-03-27 01:31:23 - MyApp]          You probably need to update the library versions.
[2016-03-27 01:31:23 - MyApp]          (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedlibraryclassmember)
[2016-03-27 01:31:23 - MyApp] java.io.IOException: Please correct the above warnings first.
[2016-03-27 01:31:23 - MyApp] 	at proguard.Initializer.execute(Initializer.java:473)
[2016-03-27 01:31:23 - MyApp] 	at proguard.ProGuard.initialize(ProGuard.java:233)
[2016-03-27 01:31:23 - MyApp] 	at proguard.ProGuard.execute(ProGuard.java:98)
[2016-03-27 01:31:23 - MyApp] 	at proguard.ProGuard.main(ProGuard.java:538)`

So, the only choice I have left is ACRA 4.8+, which does not work at the moment with using annotation. When I'm not inundated with work, I might pull the ACRA 4.8 source and see why it's not working.

**Comment by CKDPublishing** [comment]
I tried ACRA 4.8.5 with annotation and got the following error. Note: You can ignore the first part of the error as this part was done deliberately to crash the app so it can launch the ACRA error dialog:

`04-07 23:20:11.189: E/ACRA(8256): ACRA caught a RuntimeException for au.com.company.test
04-07 23:20:11.189: E/ACRA(8256): java.lang.RuntimeException: Unable to start activity ComponentInfo{au.com.company.test/au.com.company.test.MainActivity}: java.lang.NumberFormatException: Invalid int: "null"
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2658)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2723)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.access$900(ActivityThread.java:172)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1422)
04-07 23:20:11.189: E/ACRA(8256): 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-07 23:20:11.189: E/ACRA(8256): 	at android.os.Looper.loop(Looper.java:145)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.main(ActivityThread.java:5832)
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.reflect.Method.invoke(Native Method)
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.reflect.Method.invoke(Method.java:372)
04-07 23:20:11.189: E/ACRA(8256): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
04-07 23:20:11.189: E/ACRA(8256): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
04-07 23:20:11.189: E/ACRA(8256): Caused by: java.lang.NumberFormatException: Invalid int: "null"
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.Integer.invalidInt(Integer.java:138)
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.Integer.parseInt(Integer.java:358)
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.Integer.parseInt(Integer.java:334)
04-07 23:20:11.189: E/ACRA(8256): 	at au.com.company.test.MainActivity.onCreate(MainActivity.java:114)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.Activity.performCreate(Activity.java:6221)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2611)
04-07 23:20:11.189: E/ACRA(8256): 	... 10 more
04-07 23:20:11.349: E/ACRA(8256): ACRA failed to capture the error - handing off to native error reporter
04-07 23:20:11.349: E/ACRA(8256): android.content.ActivityNotFoundException: Unable to find explicit activity class {au.com.company.test/org.acra.dialog.CrashReportDialog}; have you declared this activity in your AndroidManifest.xml?
04-07 23:20:11.349: E/ACRA(8256): 	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:1777)
04-07 23:20:11.349: E/ACRA(8256): 	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1499)
04-07 23:20:11.349: E/ACRA(8256): 	at android.app.ContextImpl.startActivity(ContextImpl.java:1545)
04-07 23:20:11.349: E/ACRA(8256): 	at android.app.ContextImpl.startActivity(ContextImpl.java:1527)
04-07 23:20:11.349: E/ACRA(8256): 	at android.content.ContextWrapper.startActivity(ContextWrapper.java:337)
04-07 23:20:11.349: E/ACRA(8256): 	at org.acra.builder.ReportExecutor.dialogAndEnd(ReportExecutor.java:234)
04-07 23:20:11.349: E/ACRA(8256): 	at org.acra.builder.ReportExecutor.execute(ReportExecutor.java:222)
04-07 23:20:11.349: E/ACRA(8256): 	at org.acra.builder.ReportBuilder.build(ReportBuilder.java:148)
04-07 23:20:11.349: E/ACRA(8256): 	at org.acra.ErrorReporter.uncaughtException(ErrorReporter.java:260)
04-07 23:20:11.349: E/ACRA(8256): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:693)
04-07 23:20:11.349: E/ACRA(8256): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:690)
04-07 23:20:11.349: E/AndroidRuntime(8256): FATAL EXCEPTION: main
04-07 23:20:11.349: E/AndroidRuntime(8256): Process: au.com.company.test, PID: 8256
04-07 23:20:11.349: E/AndroidRuntime(8256): java.lang.RuntimeException: Unable to start activity ComponentInfo{au.com.company.test/au.com.company.test.MainActivity}: java.lang.NumberFormatException: Invalid int: "null"
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2658)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2723)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.access$900(ActivityThread.java:172)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1422)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.os.Looper.loop(Looper.java:145)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.main(ActivityThread.java:5832)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.reflect.Method.invoke(Native Method)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.reflect.Method.invoke(Method.java:372)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
04-07 23:20:11.349: E/AndroidRuntime(8256): Caused by: java.lang.NumberFormatException: Invalid int: "null"
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.Integer.invalidInt(Integer.java:138)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.Integer.parseInt(Integer.java:358)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.Integer.parseInt(Integer.java:334)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at au.com.company.test.MainActivity.onCreate(MainActivity.java:114)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.Activity.performCreate(Activity.java:6221)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2611)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	... 10 more
`

The interesting bit is this:

`android.content.ActivityNotFoundException: Unable to find explicit activity class {au.com.company.test/org.acra.dialog.CrashReportDialog}; have you declared this activity in your AndroidManifest.xml?`

Even though I have declared the following directives in my `AndroidManifest.xml` file:

`		<activity
			android:name="org.acra.CrashReportDialog"
	        android:theme="@style/Theme.Dialog"
        	android:launchMode="singleInstance"
        	android:excludeFromRecents="true"
        	android:finishOnTaskLaunch="true">
		</activity>`
`		<service
		    android:name="org.acra.sender.SenderService"
		    android:exported="false"
		    android:process=":acra"/>`

So, is the error complaining about the missing directive in the `ACRA 4.8.5.jar` itself?

**Comment by william-ferguson-au** [comment]
@CKDPublishing can you please create a new issue instead of hijacking this one.
Also please format any log that you post so that it is readable.

**Comment by CKDPublishing** [comment]
"@CKDPublishing can you please create a new issue instead of hijacking this one." - Sure.

**Comment by IcepaxKnight** [comment]
I have encountered same issue as the OP.

What's the ETA for the fix, please?

**Comment by william-ferguson-au** [comment]
@IcepaxKnight have you got a pull request?

**Comment by IcepaxKnight** [comment]
@William: No. How do one request a bug fix, or pull request, in Github? Elsewhere, like Google Issue Tracker, you just log an issue and it's assigned to be fixed. I'm not familiar with Girhub. Are we supposed to fix the error ourselves if we find a bug with ACRA?!

**Comment by F43nd1r** [comment]
@IcepaxKnight Acra is not backed by a big company like Google. I actually used some time on this issue, but was  unable to reproduce it. So, we rely on someone who actually has the issue to investigate it. By the way - I'm also just a user who started to help out a bit here. 

You have to understand that on a project like Acra no one guarantees stability or instant support. @william-Ferguson-au does a peetty good job though.

**Comment by william-ferguson-au** [comment]
@IcepaxKnight ACRA is an open source project that is made freely available to all by donations of time and effort from the community that use ACRA.

I co-ordinate that effort and supply what time I can afford. But I have a full time job and I also run a start up for another 40-60 hours a week.

If you are using ACRA then you are a developer. So if you'd like to add a feature or find some bug that you really want fixed, the best path is to investigate, work out the solution and create a pull request with your fix. You are free to use ACRA with your fix and we can vet your pull request  and merge it in so everyone else gets that fix too.

That's the way an open source project works.

**Comment by IcepaxKnight** [comment]
@F43nd1r: The crash dialog not displaying via annotation is reproducible on Android mobile devices running 6.0.1, 6.0, 5.0, 4.4, 4.2, 4.0.3 and 2.3.5. All you get with these Android platforms is the default crash dialog from Android. Looks like ACRA is not even started, so I don't know how you got it working.

Thanks, for the headsup. Until my employer hires more Android programmers to help me with the workload, I can't say that I have time at the moment to pull the ACRA source codes and fix this annotation bug myself. I'll come back to ACRA when I have more time in the future. Sorry, I'm just overwhelm with Android work currently.

**Comment by IcepaxKnight** [comment]
@William
> You are free to use ACRA with your fix and we can vet your pull request and merge it in so everyone else gets that fix too.

Thanks. I gather that much from the previous poster.

**Comment by Qttl** [comment]
I'm new to Android and Java so my experience is mediocre. Anyway, I came across this post as I'm having the same issue as the poster in that the crash dialog is not shown when using annotation with ACRA 4.8.5. I would like to vote- or star this bug like with Google's Issue Tracker- for it to be fixed in the next ACRA release as I can see this bug is a show-stopper, particularly for users running Android 4, 5 and 6. I have tested against these Android platforms and ACRA is not working at all on any of them. I'm not proficient yet in Java to be able to "pull" the source codes of ACRA and fix the bug with annotation. Maybe in another few months or so when I have enough experience with Android and Java, then I could try to help out with bug fixing with ACRA.

Is there any chance that this major bug can be fixed for ACRA 4.8.6 release? Or, will it be pushed back?

I'm currently evaluating which error reporting engine to use and ACRA- in alphabetical order- happened to be the first engine I'm evaluating. I read somewhere that the Android compiler is not checking for unchecked exceptions. Hence, the need for an error reporting engine like ACRA, Crittercism, etc. Totally new way of thinking for an ex-assembly programmer like myself!

**Comment by william-ferguson-au** [comment]
OK - this is **NOT** a major bug or a showstopper.

ACRA still works.
_Notification_ by dialog appears to be having an issue for some users in some environments.
If you use Toast or Silent notification then there is no issue.

**Comment by F43nd1r** [comment]
@ChuongPham @IcepaxKnight @Qttl can any of you try to replace the complete configuration in the annotation with a configuration using `ConfigurationBuilder`? `ConfigurationBuilder` should have the same options as the annotation.

**Comment by Qttl** [comment]
@William: I'm working in the pharmaceutical industry, so patient confidentiality and transparency is paramount. So, every confirmation in our Android app uses a dialog. This is a requirement from our pharmaceutical clients. Hence, using a toast or silent mode will not be accepted- even for sending error log to us by our clients- i.e. our clients want to see what information is being sent to us for troubleshooting. Hopefully, now you can understand why I stated this bug with the crash dialog is a show-stopper- at least for us here anyway.

@F43nd1r: Do you have a working example that you can post here? I'm a novice with Java and Android, so any guidance is greatly appreciated.

**Comment by F43nd1r** [comment]
@Qttl I've uploaded my testing project [here](https://drive.google.com/open?id=0B40xU-30MxN7MG1WZ25JRDV0bE0). (It uses resCertificate, which is not available in 4.8.5, because I was testing a new configuration. Just remove it.)

**Comment by Qttl** [comment]
@F43nd1r: Thanks for providing your ACRA example. Much appreciated. I have two questions regarding the example:

1. I saw you declared the following in the test app's `AmdroidManifest.xml` file:
```
        <activity
            android:name="org.acra.CrashReportDialog"
            android:theme="@style/AlertDialog.AppCompat"
            android:launchMode="singleInstance"
            android:excludeFromRecents="true"
            android:finishOnTaskLaunch="true">
        </activity>
        <service
            android:name="org.acra.sender.SenderService"
            android:exported="false"
            android:process=":acra"/>
```
Is this still required per app's declaration? Or, does ACRA already include these directives so you don't need to declare them yourself in your app's own `AndroidManifest.xml` file?

2. I have similar declaration as yours below:

`@ReportsCrashes(
        mode = ReportingInteractionMode.DIALOG,
        resDialogTitle = R.string.dialog_title,
        resDialogText = R.string.dialog_text,
        resDialogCommentPrompt = R.string.dialog_comment,
        reportSenderFactoryClasses = { LocalReportSenderFactory.class },
        buildConfigClass = BuildConfig.class,
        resCertificate = R.raw.cert
)
`
However, I don't have the `buildConfigClass` annotation. What's is in your `BuildConfig.class`? I couldn't find this class included in the zip file you've uploaded. I ignore `resCertificate` as I don't use this annotation.

My `ReportSender` and `ReportSenderFactory` classes are similar to yours.

I don't see the `ConfigurationBuilder` code block you mentioned above in one of your comments. Is the `ConfigurationBuilder` method the one you want me to test? If yes, I don't see it being declared in your test ACRA project in `App.java`.  I have looked at all the Java classes in the zip file, but can't find the code example of the `ConfigurationBuilder` method.

I look forward to your response.

**Comment by F43nd1r** [comment]
> I saw you declared the following in the test app's AmdroidManifest.xml file

No, actually they should not be there. I tested if they were the issue of this bug, because @ChuongPham has them included.

> However, I don't have the buildConfigClass annotation

BuildConfig is a generated class. It is included in every app (I think) and contains Information like if this is a debug build, or the current version. You normally don't have to specify this in your annotation, acra finds it pretty consitently. I don't remember what I was testing when I put it there.

> I don't see the ConfigurationBuilder code block you mentioned above in one of your comments.

I did not configure it with `ConfigurationBuilder`, because for me this code is working.
However, here is code which would do just that with a `ConfigurationBuilder`:
```
        ConfigurationBuilder builder = new ConfigurationBuilder(this)
                .setReportingInteractionMode(ReportingInteractionMode.DIALOG)
                .setResDialogTitle(R.string.dialog_title)
                .setResDialogText(R.string.dialog_text)
                .setResDialogCommentPrompt(R.string.dialog_comment)
                .setReportSenderFactoryClasses(LocalReportSender.class)
                .setBuildConfigClass(BuildConfig.class);
        ACRA.init(this, builder.build());
```

**Comment by Qttl** [comment]
> BuildConfig is a generated class. It is included in every app...

Yes, it is included as part of the app, so I ACRA just pick up the class from the app.

> .setReportSenderFactoryClasses(LocalReportSender.class)

I tested the `ConfigurationBuilder` via code but I have an error with the following line of code:
```
.setReportSenderFactoryClasses(LocalReportSender.class)
```
Then with:
```
.setReportSenderFactoryClasses(LocalReportSenderFactory.class)
```
In both instances, Eclipse complains first with:
```
The method setReportSenderFactoryClasses(Class<? extends ReportSenderFactory>[]) in the type ConfigurationBuilder is not applicable for the arguments (Class<LocalReportSender>)
```
Then with:
```
The method setReportSenderFactoryClasses(Class<? extends ReportSenderFactory>[]) in the type ConfigurationBuilder is not applicable for the arguments (Class<LocalReportSenderFactory>)
```
My declaration for the `ReportSender` is:
```
public class LocalReportSender implements ReportSender {
	@Override
	public void send(Context context, CrashReportData report) throws ReportSenderException {
    }
}
```
And for the `ReportSenderFactory` class is:
```
public class LocalReportSenderFactory implements ReportSenderFactory {
	@Override
	public ReportSender create(Context context, ACRAConfiguration config) {
		return new LocalReportSender();
	}
}
```
So, I'm not sure why Eclipse is complaining. Cleaning the project and recompile does not resolve the error.

Why is it complaining about the class, do you think?

**Comment by F43nd1r** [comment]
```
.setReportSenderFactoryClasses(LocalReportSender.class)
```
should be
```
 .setReportSenderFactoryClasses(new Class[]{LocalReportSenderFactory.class})
```
Sorry, I wrote this directly in here, so I had no IDE checking it.

**Comment by Qttl** [comment]
I made the changes above as you suggested, but I got just the default crash dialog from Android 5.0 with the message:
```
Unfortunately, Android Test has stopped.
```
with just an OK button. So, I don't see the ACRA crash dialog at all! Note: "Android Test" is the name of my test app. The device I'm testing this app on is Samsung Galaxy S5 running 5.0 Lollipop build.

**Comment by F43nd1r** [comment]
Are you building with gradle? If yes, could you try if it works with Android Studio? If no, can you try it with gradle?

**Comment by Qttl** [comment]
> Are you building with gradle? If yes, could you try if it works with Android Studio? If no, can you try it with gradle?
No, I don't have Android Studio, I use Eclipse and I'm building it manually (for now) via Andmore (Eclipse tool for Android- since ADT is deprecated for Eclipse). I'm using `acra-4.8.5.jar` downloaded from [here](http://search.maven.org/#search%7Cga%7C1%7Cacra). Is this the same version as the jar you're using?

I can download AS and try it with gradle. Would it make a difference, though?

**Comment by F43nd1r** [comment]
Since 4.7 ACRA requires you to use an **aar**. Please try again with the aar instead of the jar.
Gradle uses the aar by default, so this might actually be the problem here.

**Comment by Qttl** [comment]
> Since 4.7 ACRA requires you to use an aar. Please try again with the aar instead of the jar.
Gradle uses the aar by default, so this might actually be the problem here.

Yes, it could be a problem indeed if ACRA have stringent requirement for AAR instead of JAR.

Unfortunately, AAR format is not supported by Andmore, and Eclipse complains of compilation errors if I remove the ACRA JAR file from the `libs` directory.

Well, there's a few things I can only do now:
1) Ask Eclipse folks if they have any plan to include support for AAR files with Andmore future release.
2) Start learning Maven and look into Maven Eclipse's plugin so I can build with AAR file.
3) Look for other ways (if any) to build my Android project with ACRA's AAR file via Eclipse.

Other than that, I don't think I can proceed any further with this issue,

**Comment by william-ferguson-au** [comment]
If you plan on doing any Android development, then you need a build tool / environment that can consume AAR libraries. An AAR contains metadata, resources and class files in a format that can be integrated into an Android project.

You should really checkout Android Studio or at least IntelliJ for your IDE. I stopped using Eclipse in 2008/2009 and even the Android tools group stopped 2 years ago.

And you should strongly considered decoupling your build tools from your IDE. Use something industrial strength like Maven or at least Gradle.

**Comment by william-ferguson-au** [comment]
@ChuongPham @IcepaxKnight I presume your issue is not the same as @Qttl ?
Are you using an AAR in your build?

**Comment by F43nd1r** [comment]
> or at least Gradle

Gradle is the official build tool for android. It can do (almost) everything maven does and more with shorter code. If you are not already familiar with maven (or adopt a projet that already uses maven) I can't see any reason to choose Maven over Gradle for Android development.

**Comment by IcepaxKnight** [comment]
@william-ferguson-au: I was using M2Eclipse Maven plugin in Eclipse with `acra-4.8.5.aar` file. However, due to the time I spent getting the ACRA's crash dialog to work, my employer has instructed me to move on to `Crittercism`, which he found on the Internet and want me to test it out to see if it integrates without too much hassle with our Android projects and other non-Android projects.

I get paid by the hour, so I guess my employer didn't want me to stick with ACRA too long if I can't get it to work. Sorry. I have moved on, so to speak.

**Comment by ChuongPham** [comment]
@William: I'm using the M2Eclipse Maven plugin for Eclipse with the AAR file shipped with ACRA 4.8.5 release. However, I don't know if I was having issues with the plugin or that my `pom` configuration was wrong. This is the first time I have used Maven. I used Ant previously for Android builds. But since ACRA 4.7, AAR is the preferred input instead of JAR, so I started digging into Maven. Anyway, I have rolled back to ACRA 4.6.2 so I can release a new version of my client's app that fixes a few critical issues. I'll look into ACRA 4.8.5 again once I get this fix out of the way.

I'm also looking at `Buildship` Gradle plugin for Eclipse. I'm in favour of "less is more". So, if gradle can do more with less code, then this will be the path I'm taking.

**Comment by william-ferguson-au** [comment]
@ChuongPham OK, I think I see the problem. The m2eclipse plugin is solely for trying to help Eclipse comprehend a Maven config. It doesn't know anything about Android or how to build an Android app.

If you intend to use Maven to build Android app then you want to use the android-maven-plugin.

But if you are new to Android (and it sounds like you are) then I recommend using Gradle and follow the queues by the Android tools team.

And as I mentioned to @Qttl, I strongly recommend you 

1. Separate your build from your IDE
2. Migrate your IDE. Eclipse was great in the early 2000s, but it's 2016 now.

**Comment by ChuongPham** [comment]
@William: You're correct. I realised I have used a wrong type of the Maven plugin for Android. Hence, all the issues. I have since dumped `M2Eclipse` and started looking into the `Buildship` gradle plugin for Eclipse. Once I'm familiar with it, I will use it to test ACRA 4.8.5 again.

The reason I'm still using Eclipse is that many of my clients' programming teams are still using Eclipse- despite its flaws- and I need to support them with like IDE and project structure. If I send them an Android project from Android Studio, or a non-Android project not from Eclipse, they will have a fit. Clients are hard to convince, particularly if they have made a lot of investments with Eclipse. I guess I have little choice in this case.

**Comment by william-ferguson-au** [comment]
Hm. We made a decision 16 years ago to never let IDE artifacts or config get committed into our codebase. We had 2 strict rules:

1. Any developer should be able to check out the code and build it from the command line with no extra stuffing around for a new project.
2. Any developer should also be able to choose whatever IDE they like and use it to work on the code.

Any build tools that didn't support (1) was ditched. Hence our use of Maven (Gradle gets close now).
Any IDE that didn't support (2) was forbidden.

It was the best dev environment decision we ever made.

**Comment by ChuongPham** [comment]
> We made a decision 16 years ago to never let IDE artifacts or config get committed into our codebase.

Don't get me wrong, but I agree with your approach 100%.

However, if the clients pay your bills and put food on your table, one often has to "bend over backwards" to satisfy their business requirements. ;)

**Comment by F43nd1r** [comment]
> It was the best dev environment decision we ever made.

But not everybody gets to choose. I worked two months on a project which was using IntelliJ and maven. I can't just turn up there and say: Nah I don't like maven. I'll spend my first week on migrating to gradle.

BTT: In conclusion, was this issue completely client-sided?

**Comment by william-ferguson-au** [comment]
The build tool was never a single developer decision. It was a team decision.
But the team agreed that we needed a tool that would build outside of an IDE and had a minimal footprint. 

Once that was decided it meant we we free to let each developer choose their own IDE as long as they always made sure it was built and tested cleanly outside the IDE before checkin. Failure was a beerable offence.

I still want to retest what I saw originally. 
But it looks like from all 3 others it was a client side build issue.

**Comment by IcepaxKnight** [comment]
@william-ferguson-au et al:

> We made a decision 16 years ago to never let IDE artifacts or config get committed into our codebase.

To each their own. It's all come down to personal preferences. I'm a contractor and I can't just tell the client to do this instead of that. I can recommend best practices, but it's up to the client to decide what they want to do.

First golden rule of business: The client is always right, even when they are wrong.

So, if the client wants to check in IDE artifacts or config, then it's their choice. That's their mode of operation. As long as they pay me, I don't really care what they do with their business environment- including what IDE they chose. It's like me coming to Australia and tell all Australians to drive German cars instead of Holden cars. You know the Australian answer to that, right?

Anyway, ACRA was not working for me even after I dumped M2Eclipse plugin and chose a Maven plugin that works with Android for Eclipse. I tried `Crittercism` and it works with minimum hassle, so my boss decided to go with it such that he won't be considering ACRA anymore- even down the track. Sorry.
